test/integration/test_large_files.py

```diff
@@ -0,0 +1,195 @@
+
+    def write_zeros(self, file, number):
+        line = b'0' * 1000 + b'\n'
+        line_len = len(line)
+        written = 0
+        while written <= number:
+            file.write(line)
+            written += line_len
+
+    def create_bucket(self):
+        return self.b2_api.create_bucket(
+            self.generate_bucket_name(),
+            'allPublic',
+            bucket_info={BUCKET_CREATED_AT_MILLIS: str(current_time_millis())}
+        )
+
+    def test_large_file(self):
+        bucket = self.create_bucket()
+        with mock.patch.object(
+            self.info, '_recommended_part_size', new=self.info.get_absolute_minimum_part_size()
+        ):
+            bytes_to_write = int(self.info.get_absolute_minimum_part_size() * 2.5)
+            download_manager = self.b2_api.services.download_manager
+            with mock.patch.object(
+                download_manager,
+                'strategies',
+                new=[
+                    ParallelDownloader(
+                        max_streams=download_manager.DEFAULT_MAX_STREAMS,
+                        min_part_size=self.info.get_absolute_minimum_part_size(),
+                        min_chunk_size=download_manager.MIN_CHUNK_SIZE,
+                        max_chunk_size=download_manager.MAX_CHUNK_SIZE,
+                    )
+                ]
+            ):
+
+                # let's check that small file downloads fail with these settings
+                bucket.upload_bytes(b'0', 'a_single_zero')
+                with pytest.raises(ValueError) as exc_info:
+                    with io.BytesIO() as io_:
+                        bucket.download_file_by_name('a_single_zero').save(io_)
+                assert exc_info.value.args == ('no strategy suitable for download was found!',)
+
+                with TempDir() as temp_dir:
+                    temp_dir = pathlib.Path(temp_dir)
+                    source_large_file = pathlib.Path(temp_dir) / 'source_large_file'
+                    with open(source_large_file, 'wb') as large_file:
+                        self.write_zeros(large_file, bytes_to_write)
+                    bucket.upload_local_file(source_large_file, 'large_file')
+                    target_large_file = pathlib.Path(temp_dir) / 'target_large_file'
+
+                    bucket.download_file_by_name('large_file').save_to(target_large_file)
+                    assert hash_file(source_large_file) == hash_file(target_large_file)
+
+    @classmethod
+    def _should_remove_bucket(cls, bucket):
+        if bucket.name.startswith(cls.this_run_bucket_name_prefix):
+            return True
+        if bucket.name.startswith(cls.general_bucket_name_prefix):
+            if BUCKET_CREATED_AT_MILLIS in bucket.bucket_info:
+                if int(bucket.bucket_info[BUCKET_CREATED_AT_MILLIS]
+                      ) < current_time_millis() - ONE_HOUR_MILLIS:
+                    return True
+        return False
+
+    @classmethod
+    def clean_buckets(cls):
```

Comment:
I think I've seen this somewhere already, isn't this a clone?

Reply:
It's in CLI, we currently have no means of importing test code from one repo into another.